### PR TITLE
Add API version to URLs.

### PR DESCRIPTION
**Pull request by jmcarp**
Prefix API URLs with the current API version, starting with `v1`.

Note: Depends on https://github.com/18F/openFEC/pull/665

[Resolves https://github.com/18F/openFEC/issues/44]

### CODE REVIEW

**Comment by arowla** [comment]
Woooo, our Flask got a Blueprint!

**Comment by jmcarp** [comment]
:large_blue_diamond: :feet:

File: static/js/modules/typeahead.js [hunk 1]
```diff
@@ -52,8 +53,13 @@ module.exports = {
         url = "/rest/names?q=%QUERY";
 
     if (typeof API_LOCATION !== 'undefined') {
-        url = "/names?q=%QUERY&api_key=6wpqET5GjxCwAyBqXjc4dWrfMdshFE6wu3txb6yZ"; // Replace with the correct api key
-        url = API_LOCATION.concat(url);
+        url = URI(API_LOCATION)
+          .path([API_VERSION, 'names'].join('/'))
+          .query({
+            q: '%QUERY',
+            api_key: '6wpqET5GjxCwAyBqXjc4dWrfMdshFE6wu3txb6yZ'
```
Comment (msecret) [inline static/js/modules/typeahead.js]: is api_key supposed to be a constant here? Shouldn't it be reading it from config somehow?

Comment (arowla) [inline static/js/modules/typeahead.js]: Yes. This was a quick hack to get things going for the demo this past
Tuesday.

On Thu, Apr 30, 2015 at 1:11 PM, Marco Segreto <notifications@github.com>
wrote:

> In static/js/modules/typeahead.js
> <https://github.com/18F/openFEC-web-app/pull/108#discussion_r29448969>:
>
> > @@ -52,8 +53,13 @@ module.exports = {
> >          url = "/rest/names?q=%QUERY";
> >
> >      if (typeof API_LOCATION !== 'undefined') {
> > -        url = "/names?q=%QUERY&api_key=6wpqET5GjxCwAyBqXjc4dWrfMdshFE6wu3txb6yZ"; // Replace with the correct api key
> > -        url = API_LOCATION.concat(url);
> > +        url = URI(API_LOCATION)
> > +          .path([API_VERSION, 'names'].join('/'))
> > +          .query({
> > +            q: '%QUERY',
> > +            api_key: '6wpqET5GjxCwAyBqXjc4dWrfMdshFE6wu3txb6yZ'
>
> is api_key supposed to be a constant here? Shouldn't it be reading it from
> config somehow?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/18F/openFEC-web-app/pull/108/files#r29448969>.
>



-- 
Alison Rowland
Technical Lead, https://fbopen.gsa.gov | 18F
202-317-0124

**Pull request merged**